runbotics/runbotics-actions-windows/src/automation/office/excel.utils.ts

```diff
@@ -0,0 +1,42 @@
+import { CellCoordinates, GetCellCoordinatesParams } from "./excel.types";
+import ExcelErrorLogger from "./excelError.logger";
+
+export class ExcelUtils {
+    static switchWorksheet(getSession: () => any, setSession: (session: any) => void, setPrevWorksheet: (worksheet?: string) => void, worksheet?: string): void {
+        const session = getSession();
+        if (!session) return;
+        if (!worksheet || worksheet === session.ActiveSheet.Name) return;
+
+        setPrevWorksheet(session.ActiveSheet.Name);
+        session.Worksheets(worksheet).Activate();
+        setSession(session);
+    }
+
+    static switchPrevWorksheet(getSession: () => any, setSession: (session: any) => void, setPrevWorksheet: (worksheet: string | null) => void, prevWorksheet: string | null): void {
+        if (!prevWorksheet) return;
+        const session = getSession();
+        session.Worksheets(prevWorksheet).Activate();
+        setPrevWorksheet(null);
+        setSession(session);
+    }
+
+    static getCellCoordinates(getSession: () => any, { startColumn, startRow, endColumn, endRow }: GetCellCoordinatesParams): CellCoordinates {
+        return {
+            startColumn: startColumn ? this.getColumnCoordinate(getSession, startColumn) : 1,
+            startRow: startRow ?? 1,
+            endColumn: this.getColumnCoordinate(getSession, endColumn),
+            endRow: endRow ?? null
+        };
+    }
+
+    static getColumnCoordinate(getSession: () => any, column: string | number): number {
+        if (!column) return null;
+        const columnNumber = Number(column);
+        if (!isNaN(columnNumber)) return columnNumber;
+        try {
+            return getSession().ActiveSheet.Range(`${column}1`).Column;
+        } catch (e) {
+            ExcelErrorLogger.deleteColumnsIncorrectInput(e);
```

Comment:
I would use this in the insert columns actions (and I think it can be used in many places) so imo error logger should be more universal.